packages/clerk-js/src/core/clerk.ts

```diff
@@ -908,10 +914,31 @@ export default class Clerk implements ClerkInterface {
     this.#componentControls?.updateProps(props);
   };
 
+  #syncWithPrimary = () => {
+    const q = new URLSearchParams({
+      redirect_url: `${window.location.href}?synced=true`,
+    });
+    window.location.replace(new URL(`/v1/client/sync?${q.toString()}`, `https://${this.domain}`).toString());
+  };
+
   #loadInStandardBrowser = async (): Promise<void> => {
+    console.log('----- isSatellite -----', this.isSatellite);
+    if (this.isSatellite) {
+      const q = new URLSearchParams(window.location.search);
+      if (!q.has('synced')) {
```

Comment:
:upside_down_face: We can also check that `synced` has a value of true (or "true"). 

It would also be nice if we could rewrite the URL to get rid of the `synced` parameter in the query string. 

This means that we'll end up in exactly the same URL as we started. Not sure if it's doable, we can try in a follow-up PR.

Reply:
✅